app/models/workflow.rb

```diff
@@ -130,4 +138,14 @@ def concerns_subjects?
   def concerns_users?
     user_rules.present? or reducers.where(topic: 'reduce_by_user').present?
   end
+
+  def activate!
+    Workflow.transaction do
+      active!
+      pending_classification_ids.each do |id|
+        ExtractWorker.perform_async(id)
+      end
+      pending_classifications.delete_all
```

Comment:
Couldn't this delete classifications before their workers run?

Reply:
This should just delete the records from the association unless I'm misunderstanding the API docs.

Reply:
OK. That's always kind of confusing because of how it depends on what you're calling that method on (model vs association, what the `dependent` behavior is, etc), might be worth a quick test in the console.